Style for submissions/markdown

ashpil opened the pull request:
Bunch of styling added here.

See post markdown:

![image](https://user-images.githubusercontent.com/9152478/82112063-d35e6c00-9706-11ea-80f5-e77ef29c141c.png)
(I don't think different heading sizes make sense right now with the title of the actual post being the size it is)
![image](https://user-images.githubusercontent.com/9152478/82112027-a4e09100-9706-11ea-9362-52099988d216.png)

And forms/submission page:

![image](https://user-images.githubusercontent.com/9152478/82112103-14568080-9707-11ea-8a46-d9c65388cf79.png)
I finally figured out how to center that nasty checkbox!

And for the people with tinyyyy screens:

![image](https://user-images.githubusercontent.com/9152478/82112117-36500300-9707-11ea-8fc6-450bd0f82ec0.png)

ashpil commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Navbar UI bug on small screens](https://trello.com/c/ZG0wFsXY/28-navbar-ui-bug-on-small-screens)

ashpil commented:
Ooops it seems one of my changes wasn't as wise as I thought it was. I think it's because I renamed the submit button to "Submit" instead of "Create Submission", because I felt that was unnecessarily long. Figured I didn't understand Rails, but could at least figure out how to do that! Apparently not.

hernanat commented:
ahh. Capybara is getting confused @ashpil because there are two clickable things on the page with `Submit` as their text -- the `Submit` link on the navbar, and the `Submit` button in the form. What you can do to handle this is something like this:

in `spec/support/page_objects/create_submission_page.rb`:

```ruby
def submit_form
  within("form#new_create_submission_form") do
    click_on t("helpers.submit.create_submission_form.create")
  end
  self
end
```

which will scope the action to inside the form

ashpil commented on app/assets/stylesheets/pages/_submissions.scss:
```diff
@@ -32,22 +33,84 @@
   border-top: $light-border;
-  padding-top: $spacing-smaller
+  padding-top: $spacing-smaller;
+  blockquote {
```
I agree, that sounds like a good idea. If you could make the posts and comments both have a "markdown" class or something like that, it would be simple to do.

ashpil commented on app/assets/stylesheets/base/_forms.scss:
```diff
@@ -113,3 +112,17 @@ select {
   }
 }
+
+.new_create_submission_form ~ h2 {
```
Whoops, thanks for catching that! I was experimenting and ended up solving it in a different way.

ashpil commented:
This one should be good now.

hernanat commented on app/assets/stylesheets/base/_layout.scss:
```diff
@@ -22,7 +22,7 @@ body {
 }
 
-@media only screen and (max-width: 1000px) {
+@media only screen and (max-width: $mobile-cutoff) {
```
👍

ashpil commented:
Uh-oh, thought I updated the branch and squashed the commits, but apparently not. What's the proper workflow for this, just curious? Because I've been using `git rebase -i` to squash from the command line. Is there a simpler way to do this? @hernanat

hernanat commented:
what's the problem exactly? You appear to have squashed everything down into a single commit, or did you lose some changes? That is also how I squash my commits.

ashpil commented:
Well I thought I updated the branch as well before squashing, but it wants me to update it again.

hernanat commented:
ohhh I see. the changes in master are showing in your pr. Do a `git rebase origin/master`

hernanat commented:
because I merged my changes into master and your branch wasn't rebsed on top of it, it was showing those changes as part of your branch.

My workflow is typically like this:

- squash commits

- rebase on top of master

- push to remote branch

ashpil commented:
Okay, thanks! I got it now.

hernanat commented:
not a problem :)
